### PR TITLE
LPS-141065 As a form creator, I want to define validations to my date field type with other date fields in the form as parameter

### PR DESCRIPTION
**Pull request by mateusgomes01**
Related Issue: https://issues.liferay.com/browse/LPS-141065

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @mateusgomes01, can you please check if these failures are related to your changes? Thanks

**Comment by mateusgomes01** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Validation/ValidationDate.js [hunk 1]
```diff
@@ -91,36 +48,15 @@ const getDateOptionsByType = (label, name) => ({
...
-	attributeName
-) => {
-	if (localizedValue && typeof localizedValue === 'string') {
+const getSelectedParameter = (value, selectedParameterName) => {
+	if (value && typeof value === 'string') {
 		try {
-			localizedValue = JSON.parse(localizedValue);
+			value = JSON.parse(value);
 		}
 		catch (error) {}
```
Comment (miltonmc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Validation/ValidationDate.js:56]: TODO: check necessity

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Validation/ValidationDate.scss [hunk 2]
```diff
@@ -1,3 +1,7 @@
 .ddm-form-field-type__validation-date {
 	width: 33%;
 }
+
+.ddm-form-field-type__validation-date-accepted-date {
```
Comment (miltonmc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Validation/ValidationDate.scss:5]:
```suggestion
.lfr-ddm-form-field-type__validation-date-accepted-date {
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/Validation/ValidationDate.js [hunk 3]
```diff
@@ -47,6 +48,43 @@ const validations = [
...
+			quantity: 1,
+			type: 'customDate',
+			unit: 'days',
+		},
+	},
+});
+
+const parameters = generateParameter();
+
+const localizedValue = jest.fn(() => parameters['en_US']);
```
Comment (miltonmc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/Validation/ValidationDate.js:72]:
```suggestion
let localizedValue;
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/Validation/ValidationDate.js [hunk 4]
```diff
@@ -47,6 +48,43 @@ const validations = [
...
+		initialState={{builderPages, dateFieldTypeValidationEnabled}}
+		value={state}
+	>
+		<ValidationDate {...props} />
+	</FormProvider>
+);
+
 describe('ValidationDate', () => {
 	beforeAll(() => {
 		Liferay.Language.direction = {
```
Comment (miltonmc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/Validation/ValidationDate.js:90]:
```suggestion
	beforeAll(() => {
		localizedValue = jest.fn(() => parameters['en_US']);
		Liferay.Language.direction = {
```

**Review by miltonmc — APPROVED** [review]
LGTM

**Comment by miltonmc** [comment]
ci:forward

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @mateusgomes01, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @mateusgomes01, can you please check if these failures are related to your changes? Thanks

**Comment by miltonmc** [comment]
ci:test:sf

**Comment by mateusgomes01** [comment]
ci:foward

**Comment by mateusgomes01** [comment]
ci:foward

**Comment by miltonmc** [comment]
ci:forward

**Pull request closed without merging**